### PR TITLE
Add: Support for API level JWT token generation

### PR DESCRIPTION
**Pull request by BLasan**
## Purpose
Add support for API level JWT token generation in the enforcer.

## Goals

- Added new `APIPolicy` property to extract the value defined by the user.
- Added a boolean property to check whether the backend JWT generation is enabled or not.
- Added modifications in the enforcer level to generate tokens based on the value defined in API level.
- Added an integration test to check backend JWT token generation, enabled/disabled via `API Policy`

### CODE REVIEW

**Review by Krishanx92 — DISMISSED** [review]

**Comment by BLasan** [comment]
There are some refinements need to be done. Will update the PR ASAP

**Comment by BLasan** [comment]
Updated the PR, please review @Krishanx92 @AmaliMatharaarachchi

File: test/integration/integration/tests/all-http-methods-for-wildcard.go [hunk 1]
```diff
@@ -41,6 +41,7 @@ var ALLHTTPMethodsForWildCard = suite.IntegrationTest{
 		testCases := []http.ExpectedResponse{
 			// test path with trailing slash for GET
 			{
+				// TODO{Amila}: Fix issue with trailing slash #758
```
Comment (AmaliMatharaarachchi) [inline test/integration/integration/tests/all-http-methods-for-wildcard.go:44]: What is this comment, is it intentional?

Comment (BLasan) [reply test/integration/integration/tests/all-http-methods-for-wildcard.go:44]: There's an issue with this trailing slash test. Amila asked to add a comment as a TODO

**Review by AmaliMatharaarachchi — APPROVED** [review]

**Pull request merged**